Incorrect update babel-loader for webpack 1

Webpack: 1.13.2
Node: 8.9.1
npm: 5.6.0

`package.json`
```diff
{
  "devDependencies": {
+   "@babel/core": "7.0.0-beta.39",
+   "@babel/polyfill": "7.0.0-beta.39",
+   "@babel/preset-env": "7.0.0-beta.39",
+   "@babel/preset-react": "7.0.0-beta.39",
+   "@babel/preset-stage-2": "7.0.0-beta.39",
+   "babel-loader": "v8.0.0-beta.0",
-    "babel-core": "6.26.0",
-    "babel-loader": "6.2.5",
-    "babel-polyfill": "6.26.0",
-    "babel-preset-env": "1.6.1",
-    "babel-preset-react": "6.24.1",
-    "babel-preset-stage-2": "6.24.1",
  },
}
```

`.babelrc`
```diff
 {
   "presets": [
-    ["env", {
-      "targets": {
-        "browsers": ["last 2 versions"]
+    [
+      "@babel/preset-env",
+      {
+        "targets": {
+          "browsers": [
+            "last 2 versions"
+          ]
+        }
       }
-    }],
-    "react",
-    "stage-2"
+    ],
+    "@babel/preset-react",
+    "@babel/preset-stage-2"
   ]
 }
```

Error in console: 
```
ERROR in multi app
Module not found: Error: Cannot resolve module 'babel-polyfill' in /home/andrew/w/projects/m-daiquiri-ui
 @ multi app

ERROR in ./core/src/index.jsx
Module build failed: TypeError: fileSystem.statSync is not a function
    at module.exports (/home/andrew/w/projects/m-daiquiri-ui/node_modules/babel-loader/lib/utils/exists.js:5:25)
    at find (/home/andrew/w/projects/m-daiquiri-ui/node_modules/babel-loader/lib/resolve-rc.js:12:9)
    at Object.module.exports (/home/andrew/w/projects/m-daiquiri-ui/node_modules/babel-loader/lib/index.js:121:132)
 @ multi app

```

--- a/__tests__/packageData.test.js
+++ b/__tests__/packageData.test.js
@@ -3,6 +3,7 @@ const upgradeDeps = require('../src/upgradeDeps');
 const babelCoreFixture = require('../fixtures/babel-core');
 const jestFixture = require('../fixtures/jest');
 const depsFixture = require('../fixtures/deps');
+const webpackV1Fixture = require('../fixtures/webpack-v1');
 const depsFixtureEarlierBeta = require('../fixtures/deps-earlier-beta.json');
 const scriptsMochaFixture = require('../fixtures/scripts-mocha');
 const scriptsBabelNodeFixture = require('../fixtures/scripts-babel-node');
@@ -68,4 +69,8 @@ test('@babel/core peerDep', async () => {
 
 test('jest babel-core bridge', async () => {
   expect(await updatePackageJSON(jestFixture)).toMatchSnapshot();
+});
+
+test('webpack v1 compatibility', async () => {
+  expect(await updatePackageJSON(webpackV1Fixture)).toMatchSnapshot();
 });

--- a/fixtures/webpack-v1.json
+++ b/fixtures/webpack-v1.json
@@ -1,0 +1,9 @@
+{
+  "name": "webpack-v1-compatibility",
+  "devDependencies": {
+    "babel-loader": "^7.1.2",
+    "babel-plugin-transform-object-rest-spread": "^6.20.2",
+    "babel-preset-es2015": "^6.18.0",
+    "webpack": "^1.15.0"
+  }
+}

--- a/src/index.js
+++ b/src/index.js
@@ -64,20 +64,11 @@ async function updatePackageJSON(pkg) {
   };
 
   if (pkg.devDependencies) {
-    pkg.devDependencies = upgradeDeps(
+    pkg.devDependencies = sortKeys(upgradeDeps(
       pkg.devDependencies,
       getLatestVersion(),
       upgradeDepOptions,
-    );
-
-    const devDeps = Object.keys(pkg.devDependencies);
-    // use babel-bridge for jest
-    // maybe should do this for other tools?
-    if (devDeps.includes("jest") && !devDeps.includes("babel-core")) {
-      pkg.devDependencies["babel-core"] = "^7.0.0-bridge.0";
-    }
-
-    pkg.devDependencies = sortKeys(pkg.devDependencies);
+    ));
   }
 
   if (pkg.dependencies) {

--- a/src/upgradeDeps.js
+++ b/src/upgradeDeps.js
@@ -7,8 +7,6 @@ const otherPackages = {
 };
 
 module.exports = function upgradeDeps(dependencies, version, options = {}) {
-  const { hasFlow } = options;
-
   for (let pkg of Object.keys(dependencies)) {
     const depVersion = dependencies[pkg];
     if (Object.keys(oldPackages).includes(pkg)) {
@@ -53,10 +51,31 @@ module.exports = function upgradeDeps(dependencies, version, options = {}) {
     dependencies['@babel/core'] = version;
   }
 
+  const webpack = semver.coerce(dependencies.webpack);
+  const depsWebpack1 = webpack && webpack.major === 1;
+
   // Adds preset-flow if needed, especially since it was split out of
   // preset-react
-  if (hasFlow && !dependencies['@babel/preset-flow']) {
+  if (options.hasFlow && !dependencies['@babel/preset-flow']) {
     dependencies['@babel/preset-flow'] = version;
+  }
+
+  // Later versions of babel-loader are incompatible with Webpack v1.
+  // https://github.com/babel/babel-loader/issues/505
+  if (depsWebpack1 && dependencies['babel-loader']) {
+    console.log('Updating babel-loader to v7.1.1 as this project uses Webpack v1');
+    dependencies['babel-loader'] = '7.1.1';
+  }
+
+  // babel-bridge is needed for Jest, or for when a project is using Webpack v1
+  // and babel-loader.
+  // https://github.com/babel/babel-upgrade/issues/29
+  // https://github.com/babel/babel-loader/issues/505
+  if (
+    (dependencies['jest'] || (depsWebpack1 && dependencies['babel-loader'])) &&
+    !dependencies['babel-core']
+  ) {
+    dependencies['babel-core'] = '^7.0.0-bridge.0';
   }
 
   return dependencies;

